hammer/library/aws/elasticsearch.py

```diff
@@ -0,0 +1,356 @@
+        :return: 
+        """
+        log_groups = cw_client.describe_log_groups()
+
+        log_group_arn = None
+        for log_group in log_groups["logGroups"]:
+            log_group_name = log_group["logGroupName"]
+            if log_group_name == domain_log_group_name:
+                log_group_arn = log_group["arn"]
+
+        if log_group_arn:
+            """
+            In order to successfully deliver the logs to your CloudWatch Logs log group, 
+            Amazon Elasticsearch Service (AES) will need access to two CloudWatch Logs API calls:
+                1. CreateLogStream: Create a CloudWatch Logs log stream for the log group you specified
+                2. PutLogEvents: Deliver CloudTrail events to the CloudWatch Logs log stream
+
+            Adding resource policy that grants above access.
+            """
+            policy_name = "AES-"+domain_log_group_name+"-Application-logs"
+            policy_doc = {}
+            statement = {}
+            principal = {}
+            action = []
+            principal["Service"] = "es.amazonaws.com"
+            action.append("logs:PutLogEvents")
+            action.append("logs:CreateLogStream")
+            statement["Effect"] = "Allow"
+            statement["Principal"] = principal
+            statement["Action"] = action
+            statement["Resource"] = log_group_arn
+
+            policy_doc["Statement"] = statement
+
+            cw_client.put_resource_policy(
+                policyName=policy_name,
+                policyDocument=str(json.dumps(policy_doc))
+            )
+        return log_group_arn
+
+    @staticmethod
+    def set_domain_logging(es_client, cw_client, domain_name):
+        """
+        
+        :param es_client: elastic search boto3 client
+        :param cw_client: cloudwatch logs boto3 client
+        :param domain_name: elastic search domain name
+        :return: 
+        """
+        domain_log_group_name = "/aws/aes/domains/" + domain_name + "/application-logs"
+        log_group_arn = ElasticSearchOperations.retrieve_loggroup_arn(cw_client, domain_log_group_name)
+        if not log_group_arn:
+            cw_client.create_log_group(logGroupName=domain_log_group_name)
+            log_group_arn = ElasticSearchOperations.retrieve_loggroup_arn(cw_client, domain_log_group_name)
+
+        es_client.update_elasticsearch_domain_config(
+            DomainName=domain_name,
+            LogPublishingOptions={
+                'ES_APPLICATION_LOGS':
+                {
+                    'CloudWatchLogsLogGroupArn': log_group_arn,
+                    'Enabled': True
+                }
+            }
+        )
+
+    @classmethod
+    def validate_access_policy(cls, policy_details):
+        """
+
+        :param policy_details: 
+        :return: 
+        """
+        public_policy = False
+        for statement in policy_details.get("Statement", []):
+            effect = statement['Effect']
```

Comment:
I think we can reuse code from S3 operations. (and them move them out from s3 because we can use them for multiple security features)